aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -1518,6 +1518,106 @@ static void scatter_reduce_exclude_self_helper(
   });
 }
 
+void _scatter_via_index_put(
+  const Tensor& self,
+  int64_t dim,
+  const Tensor& index,
+  const Tensor& src,
+  const Tensor& mut_out) {
+  if (self.dim() == 1) {
+    // TODO: Pretty sure these checks can be removed, since they're done in
```

Comment:
can we take care of this TODO?

Reply:
let me do it in the next PR, will need to modify this function a bit.